geoutils/georaster/raster.py

```diff
@@ -817,14 +813,142 @@ def is_modified(self) -> bool:
+
+        If you wish to set nodata value without updating the mask, run this function with the update_mask argument as
+        False.
+
+        :param nodata: Nodata values
+        :param update_array: Update the old nodata values into new nodata values in the data array
+        :param update_mask: Update the old mask by unmasking old nodata and masking new nodata (if array is updated,
+            old nodata are changed to new nodata and thus stay masked)
+        """
+        if nodata is not None and not isinstance(nodata, (list, int, float, np.integer, np.floating)):
+            raise ValueError("Type of nodata not understood, must be list or float or int")
+
+        elif (isinstance(nodata, (int, float, np.integer, np.floating))) and self.count > 1:
+            print("Several raster band: using nodata value for all bands")
+            nodata = [nodata] * self.count
+
+        elif isinstance(nodata, list) and self.count == 1:
+            print("Only one raster band: using first nodata value provided")
+            nodata = list(nodata)[0]
+
+        elif nodata is None:
+            self._nodata = None
+            return
+
+        # Check that nodata has same length as number of bands in self
+        if isinstance(nodata, list):
+            if len(nodata) != self.count:
+                raise ValueError(f"Length of nodata ({len(nodata)}) incompatible with number of bands ({self.count})")
+            # Check that nodata value is compatible with dtype
+            for k in range(len(nodata)):
+                if not rio.dtypes.can_cast_dtype(nodata[k], self.dtypes[k]):
+                    raise ValueError(f"nodata value {nodata[k]} incompatible with self.dtype {self.dtypes[k]}")
+        elif isinstance(nodata, (int, float, np.integer, np.floating)):
+            if not rio.dtypes.can_cast_dtype(nodata, self.dtypes[0]):
+                raise ValueError(f"nodata value {nodata} incompatible with self.dtype {self.dtypes[0]}")
+
+        # If we update mask or array, get the masked array
+        if update_array or update_mask:
+
+            # Extract the data variable, so the self.data property doesn't have to be called a bunch of times
+            imgdata = self.data
+
+            # Loop through the bands
+            for i, new_nodata in enumerate(nodata if isinstance(nodata, Iterable) else [nodata]):
+
+                # Get the index of old nodatas
+                index_old_nodatas = imgdata.data[i, :, :] == self.nodata
+
+                # Get the index of new nodatas
+                index_new_nodatas = imgdata.data[i, :, :] == new_nodata
+
+                if np.count_nonzero(index_new_nodatas) > 0:
+                    if update_array and update_mask:
+                        warnings.warn(
+                            message="New nodata value already found in the data array, the corresponding grid cells "
+                            "will be indistinguishable from that updated from the old nodata value, and will "
```

Comment:
I found it a bit difficult to read. I would make it more consistent with the last one and say "the corresponding grid cells will be masked and the values updated to match the new nodata."
Same for the second warning.

Reply:
Actually that statement doesn't work: the values of these grid cell will not be updated. They simply will have the same value as the other cells updated from old nodata to new nodata.

I'll change the statement to this which maybe sounds easier to grasp:
"New nodata value found in the data array. Those will be masked, and the old nodata cells will now take the same value. Use set_nodata() with update_array=False and/or update_mask=False to change this behaviour`

Reply:
👍